### ISSUE TITLE
Login configuration

### ISSUE THREAD

**Issue by joshuawilson** [fabric8-ui/fabric8-runtime-console#87]
Setup the library to use the ngx-login-client and the auth that Platform is using.

**Comment by jstrachan** [comment]
so I'm thinking we either:

(i) refactor the login stuff in fabric8-runtime-console into a service that we can opt out of in fabric8-ui and just use ngx-login-client directly or

(ii) refactor the core parts of fabric8-runtime-console out as an npm module (or separate repo) and let the login stuff sit in fabric8-ui

(iii) have 2 separate components that extend fabric8-runtime-console; one that uses vanilla OAuth in OpenShift; the other that uses ngx-login-client

As we need fabric8-runtime-console that runs on vanilla Kubernetes/OpenShift without using ngx-login-client so that we can just talk OAuth to OpenShift/Keycloak.

I'm guessing ngx-login-client can't handle the case where fabric8-runtime-console is running as a client directly on OpenShift with no planner/core API backend right?

**Comment by joshuawilson** [comment]
Or we can update the login-client to handle generic oauth

**Comment by jstrachan** [comment]
yeah - current plan is I think to patch ngx-login-client so it can be configured to either use generic OAuth or use KeyCloak/Almighty stuff

**Comment by jstrachan** [comment]
lets separate out the "make ngx-login-client support vanilla OAuth" issue https://github.com/fabric8-ui/ngx-login-client/issues/20

then this issue becomes mostly 'migrate to ngx-login-client"
